ext/standard/mail.c

```diff
@@ -57,7 +57,7 @@
 
 extern zend_long php_getuid(void);
 
-static bool php_mail_build_headers_check_field_value(zval *val)
+static bool php_mail_build_headers_check_field_value(zval *val, php_mail_header_value_error_type *error_type)
```

Comment:
What about returning the error type instead of zend_result? This simplifies the API a bit.
I mean: `static php_mail_header_value_error_type php_mail_build_headers_check_field_value(zval *val)`. And you can add a `NO_HEADER_ERROR` enum value and return that instead of `SUCCESS`.

Reply:
yeah agree with Niels. Single return value would be better.

Reply:
Fixed it!